### DISCUSSION TITLE
Offset not working

### DISCUSSION THREAD

**Discussion by LuginiAndrea** [enable3d/enable3d#152] [Help / Questions]
Hello, I updated the versio of the library (went from dev.1.0 to dev.2.0) and suddenly the offset property for compound shapes 
stopped working.

Code before update: 
 ` const compounds = [{shape:"box", width: 2.9, depth: 5.5, height: 8.5,  y: -10 }];
`

and after update: 
`const compounds = [{shape:"box", width: 2.9, depth: 5.5, height: 8.5,  offset: {y: -10} }];`

Update: Found a way to make it work when I have only 1 compound shape, but I can't find a way to make it work with 2 or more compound shapes

**Comment by yandeu** [comment]
> Update: Found a way to make it work when I have only 1 compound shape, but I can't find a way to make it work with 2 or more compound shapes

Does everything work now? Or do you still need help?
Can you share a code snippet?

**Reply by LuginiAndrea** [reply]
Unfortunately no, it still doesn't work

This is my snippet:
function addTree1(scene, name, {x = 0, y = 0, z = 0} = {}) {
  const compounds = [ {shape:"box", width: 6, depth: 6, height: 24} ];
  const model = LoadModel({ scene:scene,
                            modelName:"tree1",
                            radX: Math.PI * 3/2,
                          });
  names["Tree1"]++;
  const physConf = {compounds:compounds, offset: {y:-12}, mass:Number.MAX_SAFE_INTEGER};
  return new PhysEntity({parent:scene, name:name + names["Tree1"], model:model,x:x,y:y,z:z, physConfig:physConf});
}

And what is called in the physEntity class:
constructor({parent, name, model, x = 0, y = 0, z = 0, collisionFlag = 0, physConfig = {compounds = null, mass = 1, offset = {x = 0, y = 0, z = 0} = {}} = {}, maxHealth = 0, health = 0, inventory = null} = {}) { 
        super(parent,name,model,x,y,z);
        this.__maxHealth = maxHealth; //float
        this._health = health; //float
        //Physics stuff
        this.__collisionFlag = collisionFlag;
        this.__physConfig = physConfig;
    }    

load() {
        super.load(); //Simple mesh load
        const physConf = this.__physConfig;
        this.__parent.physics.add.existing(this.model, {compound:physConf.compounds, offset:physConf.offset, mass:physConf.mass});
        this.__model.body.setCollisionFlags(this.__collisionFlag);
    }
    
    If I put an offset in the compound body it doesn't work and the offset in PhysConf is the only one that works

**Comment by NovemberDev** [comment]
Hello everyone,

I can't get this to work as well. I'm adding a 3D model and I want the sphere to be placed at half the size of the model.
For testing purposes, I attach a big sphere to it while I kind of "freeze" it's linear velocity so it doesn't fall too quick. 
It is stuck at 0,0,0 regardless of the offset. The offset gets passed down and can be seen with console.log(this.model.body.offset):
{ x: 0, y: 10.35, z: 0 }

This is my code:

```
   constructor(scene, data, material) {
    this.root = new Object3D();

    this.model = getSceneChild(data, "rig");

    this.root.add(this.model);
    Physics.ammo.add.existing(this.model, {
      shape: "sphere",
      radius: 2.65,
      offset: { y: 10.35 },
      collisionFlags: 0,
    });
    this.model.body.setLinearFactor(0.0001, 0.0001, 0.0001);
    
    // Physics.ammo comes from a custom class Physics.initialize(this.physics) which is called inside "MainScene extends Scene3D" for global access
```
   
How can I add an offset?

**Reply by yandeu** [reply]
Here is an great example:
https://youtu.be/j6nv3JIAFLk?t=703

**Reply by NovemberDev** [reply]
> Here is an great example:
> https://youtu.be/j6nv3JIAFLk?t=703

I understood the concept wrong: the offset is not for the shape - it is for the model, so the offset moves the model and not the shape...
That helped a lot, thank you!!
